dynamometer-infra/src/main/java/com/linkedin/dynamometer/SimulatedDataNodes.java

```diff
@@ -0,0 +1,153 @@
+    }
+    String nameNodeAdr = defaultFS.getAuthority();
+    if (nameNodeAdr == null) {
+      printUsageExit("No NameNode address and port in config");
+    }
+    System.out.println("DataNodes will connect to NameNode at " + nameNodeAdr);
+
+    System.setProperty(MiniDFSCluster.PROP_TEST_BUILD_DATA,
+        DataNode.getStorageLocations(getConf()).get(0).getFile().getAbsolutePath());
+    SimulatedMultiStorageFSDataset.setFactory(getConf());
+    getConf().setLong(SimulatedMultiStorageFSDataset.CONFIG_PROPERTY_CAPACITY, STORAGE_CAPACITY);
+
+    UserGroupInformation.setConfiguration(getConf());
+    MiniDFSCluster mc = new MiniDFSCluster();
+    try {
+      mc.formatDataNodeDirs();
+    } catch (IOException e) {
+      System.out.println("Error formatting DataNode dirs: " + e);
+      System.exit(1);
+    }
+
+    try {
+      System.out.println("Found " + blockListFiles.size() + " block listing files; launching DataNodes accordingly.");
+      mc.startDataNodes(getConf(), blockListFiles.size(), null, false, StartupOption.REGULAR,
+          null, null, null, null, false, true, true, null);
+      long startTime = Time.monotonicNow();
+      System.out.println("Waiting for DataNodes to connect to NameNode and init storage directories.");
+      Set<DataNode> datanodesWithoutFSDataset = new HashSet<>(mc.getDataNodes());
+      while (!datanodesWithoutFSDataset.isEmpty()) {
+        Iterator<DataNode> iter = datanodesWithoutFSDataset.iterator();
+        while (iter.hasNext()) {
+          if (DataNodeTestUtils.getFSDataset(iter.next()) != null) {
+            iter.remove();
+          }
+        }
+        Thread.sleep(100);
+      }
+      System.out.println("Waited " + (Time.monotonicNow() - startTime) + " ms for DataNode FSDatasets to be ready");
+
+      for (int dnIndex = 0; dnIndex < blockListFiles.size(); dnIndex++) {
+        Path blockListFile = blockListFiles.get(dnIndex);
+        try (FSDataInputStream fsdis = blockListFile.getFileSystem(getConf()).open(blockListFile)) {
+          BufferedReader reader = new BufferedReader(new InputStreamReader(fsdis));
+          List<Block> blockList = new ArrayList<>();
+          int cnt = 0;
+          for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+            // Format of the listing files is blockID,blockGenStamp,blockSize
+            String[] blockInfo = line.split(",");
+            blockList.add(new Block(Long.parseLong(blockInfo[0]),
+                Long.parseLong(blockInfo[2]), Long.parseLong(blockInfo[1])));
+            cnt++;
+          }
+          try {
+            mc.injectBlocks(dnIndex, blockList, bpid);
+          } catch (IOException ioe) {
+            System.out.printf("Error injecting blocks into DataNode %d for block pool %s: %s%n", dnIndex, bpid,
+                ExceptionUtils.getFullStackTrace(ioe));
+          }
+          System.out.printf("Injected %d blocks into DataNode %d for block pool %s%n", cnt, dnIndex, bpid);
```

Comment:
Looks like even if `mc.injectBlocks` throws exception, this line still gets printed out, after the error message. how about moving this line to right after `mc.injectBlocks`